distage/distage-framework-docker/src/main/scala/izumi/distage/docker/healthcheck/chain/PostgreSqlProtocolCheck.scala

```diff
@@ -0,0 +1,73 @@
+package izumi.distage.docker.healthcheck.chain
+
+import java.net.{InetSocketAddress, Socket}
+import java.nio.ByteBuffer
+
+import izumi.distage.docker.Docker.DockerPort
+import izumi.distage.docker.DockerContainer
+import izumi.distage.docker.healthcheck.ContainerHealthCheck
+import izumi.distage.docker.healthcheck.ContainerHealthCheck.HealthCheckResult
+import izumi.logstage.api.IzLogger
+import org.apache.commons.codec.binary.Hex
+
+final class PostgreSqlProtocolCheck[Tag](
+  userName: String = "postgres",
+  databaseName: String = "postgres",
+  port: DockerPort = DockerPort.TCP(5432),
+) extends ChainedContainerHealthCheck[Tag] {
+  override def check(
+    logger: IzLogger,
+    container: DockerContainer[Tag],
+    portStatus: HealthCheckResult.AvailableOnPorts,
+  ): ContainerHealthCheck.HealthCheckResult = {
+    portStatus.availablePorts.availablePorts.get(port) match {
+      case Some(value) if portStatus.requiredPortsAccessible =>
+        val startupMessage = genStartupMessage()
+        val socket = new Socket()
+        try {
+          val availablePort = value.head
+          socket.connect(new InetSocketAddress(availablePort.hostV4, availablePort.port), container.containerConfig.portProbeTimeout.toMillis.toInt)
+          logger.info(s"Checking PostgreSQL protocol on $port. Startup message: ${Hex.encodeHexString(startupMessage)}.")
```

Comment:
You don't apache commons for this, hex conversion is trivial.

Reply:
Done. I've added a small `toHex` function and place it into the `IzString`.